crates/rpc/rpc/src/validation.rs

```diff
@@ -78,6 +80,10 @@ pub struct ValidationApiInner<Provider: ChainSpecProvider, E> {
     executor_provider: E,
     /// Set of disallowed addresses
     disallow: HashSet<Address>,
+    /// Cached state reads to avoid redundant disk I/O across multiple validation attempts.
+    /// Maps block hash to cached reads for that block's state.
+    /// Uses async `RwLock` to safely handle concurrent validation requests.
+    cached_states: Arc<RwLock<HashMap<B256, CachedReads>>>,
```

Comment:
I see, ValidationApiInner is clone, but it shouldn't be.
we can remove the clone impl and use Rwlock directly.

Reply:
Ok I try to remove a couple of `Clone`